src/connections.jl

```diff
@@ -0,0 +1,173 @@
+abstract type AbstractAffineConnection{𝔽,TM<:Manifold{𝔽}} end
+
+"""
+    apply_operator(F::AbstractAffineConnection, p, X[, backend::AbstractRiemannianDiffBackend])
+
+Apply connection `F` at point `p` to vector field `Y` in direction `X`.
+"""
+function apply_operator(
+    F::AbstractAffineConnection,
+    p,
+    X,
+    Y,
+    backend::AbstractRiemannianDiffBackend,
+)
+    Z = allocate_result(base_manifold(F), apply_operator, p, X)
+    return apply_operator!(F, Z, p, X, Y, backend)
+end
+
+function apply_operator(F::AbstractAffineConnection, p, X, Y)
+    return apply_operator(F, p, X, Y, rdifferential_backend())
+end
+
+
+function apply_operator!(F::AbstractAffineConnection, Z, p, X, Y)
+    return apply_operator!(F, Z, p, X, Y, rdifferential_backend())
+end
+
+"""
```

Comment:
I think previously we always collected types atop?
And – is this meant to work similar to a metric? Ah, no, it has `M` inside, so we do not need another decorator. Looking forward to see one or two examples how this will work :)

Reply:
Right, I'll move the type to the top.